server/tests/src/main/scala/sttp/tapir/server/tests/ServerCookieHeadersTests.scala

```diff
@@ -0,0 +1,38 @@
+package sttp.tapir.server.tests
+
+import cats.implicits.catsSyntaxEitherId
+import org.scalatest.matchers.should.Matchers._
+import sttp.client3._
+import sttp.model._
+import sttp.model.headers.CookieWithMeta
+import sttp.monad.MonadError
+import sttp.tapir._
+import sttp.tapir.tests.Test
+
+class ServerCookieHeadersTests[F[_], OPTIONS, ROUTE](createServerTest: CreateServerTest[F, Any, OPTIONS, ROUTE])(implicit m: MonadError[F]) {
+
+  import createServerTest._
+
+  def tests(): List[Test] = requestTests
+
+  val requestTests = List(
```

Comment:
**suggestion:** Since this is ZIO-specific, it could go directly to `ZioHttpServerTest.additionalTests()` and you won't need the `cookieHeaders` flag.

Reply:
I though it might be good to do it this way since it does not only reproduce the ZIO-specific bug but also guard consistency among different backends' behaviour. Disclaimer: I do not have a strong opinion on this, I just want to put a spotlight on my initial motivation.

Reply:
yeah let's test it in all backends, as we want all backends to behave this way (separating the headers)

Reply:
Sorry, I didn't notice the flag is always `true`, so this means this test is executed for all the servers, right? In that case, I second Adam's opinion that the flag could be removed. And disregard my comment about moving the test to `ZioHttpServerTest`.